controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1855,6 +1855,23 @@ func (r *BareMetalHostReconciler) createHostFirmwareSettings(info *reconcileInfo
 	return nil
 }
 
+func (r *BareMetalHostReconciler) createHostUpdatePolicy(info *reconcileInfo) error {
+	// NOTE(janders) the goal here is to ensure that the controller reads the hup resource and adds OwnerReference to it
+	hup := &metal3api.HostUpdatePolicy{}
+	if !ownerReferenceExists(info.host, hup) {
```

Comment:
You're working with an empty object, you need to add loading (but not creating it).

Reply:
Fair point, now it feels I was a bit trigger happy deleting code from previous revision. Re-added the loading part, let's discuss if the code is good now.